request.go

```diff
@@ -271,22 +283,20 @@ func (ro RequestOptions) proxySettings(req *http.Request) (*url.URL, error) {
 
 }
 
-// useDefaultClient will tell the "client creator" if a custom client is needed
-// it checks the following items (and will create a custom client of these are)
-// true
-// 1. Do we want to accept invalid SSL certificates?
-// 2. Do we want to disable compression?
-// 3. Do we want a custom proxy?
-func (ro RequestOptions) dontUseDefaultClient() bool {
-	return ro.InsecureSkipVerify == true ||
-		ro.DisableCompression == true ||
-		len(ro.Proxies) != 0
-}
-
 func buildHTTPClient(ro *RequestOptions) *http.Client {
-	// Does the user want to change the defaults?
-	if !ro.dontUseDefaultClient() {
-		return http.DefaultClient
```

Comment:
You read my mind :)! I plan on moving this function to a utility (I need it for the session functionality) and it could use a new name :) (it is a product of late night coding ;))